Bottom: d1b8084a71b35861ddc6baeb54c30dbfa3d1c2d3
Top:    1f2c1378491ac9502a69cdfea0e92aa73661d60b
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-12-12 10:19:02 +0800

Refresh of macvtap-submit-skb-list-for-0

---

diff --git a/drivers/net/macvtap.c b/drivers/net/macvtap.c
index dbd093a..35dd144 100644
--- a/drivers/net/macvtap.c
+++ b/drivers/net/macvtap.c
@@ -686,10 +686,18 @@ static int macvtap_batch_xmit(struct macvtap_queue *q, struct sk_buff *skb,
 	spin_unlock(&queue->lock);
 
 	if (rcv) {
+		struct sk_buff *head = NULL;
+		struct sk_buff *tmp;
 		while ((skb = __skb_dequeue(&process_queue))) {
-			skb->dev = q->vlan->lowerdev;
-			skb_direct_xmit(skb, skb->next != NULL);
+			if (!head)
+				head = tmp = skb;
+			else {
+				tmp->next = skb;
+				tmp = skb;
+			}
 		}
+		head->dev = q->vlan->lowerdev;
+		skb_direct_xmit(head, false);
 	}
 
 	return 0;
